services/consensuscontext/validate_transactions_block_test.go

```diff
@@ -85,23 +83,167 @@ func TestTransactionsBlockValidators(t *testing.T) {
 		require.Equal(t, ErrMismatchedPrevBlockHash, errors.Cause(err), "validation should fail on incorrect prev block hash", err)
 	})
 
-	t.Run("should return error for transaction block with failing tx ordering validation", func(t *testing.T) {
-		vctx := toTxValidatorContext(cfg)
-		vctx.txOrderValidator = falsyValidateTransactionOrdering
-		err := validateTxTransactionOrdering(context.Background(), vctx)
-		require.Equal(t, ErrFailedTransactionOrdering, errors.Cause(err), "validation should fail on failing tx ordering validation", err)
-	})
-
 	t.Run("should return error for invalid timestamp of block", func(t *testing.T) {
 		vctx := toTxValidatorContext(cfg)
 		err := validateTxTransactionsBlockTimestamp(context.Background(), vctx)
 		require.Equal(t, ErrInvalidBlockTimestamp, errors.Cause(err), "validation should fail on invalid timestamp of block", err)
 	})
+}
 
+func TestConsensusContextValidateTransactionsBlockTriggerTransactionNotForwardedToPreOrder(t *testing.T) {
+	test.WithContext(func(ctx context.Context) {
+		block := builders.BlockPairBuilder().Build()
+		cfg := config.ForConsensusContextTests(nil, true)
+		txPool := &services.MockTransactionPool{}
+		txPool.When("ValidateTransactionsForOrdering", mock.Any, mock.Any).Call(func(ctx context.Context, input *services.ValidateTransactionsForOrderingInput) {
+			require.True(t, len(input.SignedTransactions) == len(block.TransactionsBlock.SignedTransactions)-1)
+		}).Return(nil, nil)
+		s := &service{txPool, nil, nil, cfg, nil, nil}
```

Comment:
this is dangerous, as you might be failing to initialize stuff that the service assumes is initialized in the `NewConcensusContext` factory func. 

why are you not using the harness? why are you initializing the service only partially? seems to me that `validateTxTransactionOrdering` needs to have a unit test without the service at all